### PR TITLE
Add an E-ink friendly trail type

### PR DESCRIPTION
**Pull request by ulipo**
Brief summary of the changes
----------------------------
Add an E-ink friendly trail type

Related issues and discussions
------------------------------
https://github.com/XCSoar/XCSoar/issues/216#issuecomment-1445405961

### CODE REVIEW

File: src/Look/TrailLook.cpp [hunk 1]
```diff
@@ -107,7 +123,7 @@ TrailLook::Initialise(const TrailSettings &settings)
     else
       iwidth = std::max(minwidth,
                         (i - NUMSNAILCOLORS / 2) *
-                        Layout::ScalePenWidth(16u) / NUMSNAILCOLORS);
+                        Layout::ScalePenWidth(8u) / NUMSNAILCOLORS);
```
Comment (lordfolken) [inline src/Look/TrailLook.cpp:126]: Is this change intended? This would affect all trails, no?

File: NEWS.txt [hunk 2]
```diff
@@ -94,10 +95,10 @@ Version 7.24 - 2022/07/22
   - rework menu labels
   - increase popup message timeouts
   - fix missing wind arrow InfoBox updates
-* file handling
```
Comment (MaxKellermann) [inline NEWS.txt:97]: Whitespace change?

File: NEWS.txt [hunk 3]
```diff
@@ -94,10 +95,10 @@ Version 7.24 - 2022/07/22
   - rework menu labels
   - increase popup message timeouts
   - fix missing wind arrow InfoBox updates
-* file handling 
+* file handling
   - guard against file corruption in poweroff situations
 * FLARM
-  - timeouts in flight downloads
```
Comment (MaxKellermann) [inline NEWS.txt:100]: Whitespace change?

File: NEWS.txt [hunk 4]
```diff
@@ -113,7 +114,7 @@ Version 7.24 - 2022/07/22
   - new default fonts
   - use 6 devices in serial port list (like all other targets)
 * XCVario/Borgelt
-  - revert cruise/circling mode change
```
Comment (MaxKellermann) [inline NEWS.txt:116]: Whitespace change?

File: src/Look/TrailLook.cpp (hunk 1 above)
Comment (MaxKellermann) [reply src/Look/TrailLook.cpp:126]: It would. That's not acceptable.

Comment (ulipo) [reply src/Look/TrailLook.cpp:126]: Well, it was intended for my personal use, I forgot to remove it from the commit.
I'll take care of that ASAP

File: NEWS.txt (hunk 2 above)
Comment (ulipo) [reply NEWS.txt:97]: I guess my editor (Gedit) stripped some trailing spaces, I only added row 14...

File: NEWS.txt (hunk 3 above)
Comment (ulipo) [reply NEWS.txt:100]: Trailing space removed, as before

File: NEWS.txt (hunk 4 above)
Comment (ulipo) [reply NEWS.txt:116]: and trailing whitespace here too...

File: src/Look/TrailLook.cpp (hunk 1 above)
Comment (ulipo) [reply src/Look/TrailLook.cpp]: ok, reverted the file to its original status.

File: NEWS.txt (hunk 2 above)
Comment (MaxKellermann) [reply NEWS.txt:97]: Then please fix your editor. If you want to add one line, then the commit diff should only show that one line, and not do lots of whitespace changes.

Comment (ulipo) [reply NEWS.txt]: Ok, done.
I really can't find any settings in Gedit about that behaviour, I think I'll file a bug.

Comment (MaxKellermann) [reply NEWS.txt]: If you want to remove trailing spaces, that would be a reasonable thing to do (probably was added accidently by somebody many years ago), but that just shouldn't be hidden inside an unrelated commit.

File: src/MapSettings.hpp [hunk 5]
```diff
@@ -97,6 +97,7 @@ struct TrailSettings {
   enum class Type: uint8_t {
     VARIO_1,
     VARIO_2,
+    VARIO_KOBO,
```
Comment (MaxKellermann) [inline src/MapSettings.hpp:100]: Adding this in the middle breaks all existing configurations, doesn't it?

File: src/Look/TrailLook.cpp [hunk 6]
```diff
@@ -54,6 +54,20 @@ GetVario2Color(short ramp_h) {
...
 }
 
+static RGB8Color
+GetVarioKoboColor(short ramp_h) {
+  static constexpr ColorRamp snail_colors_vario_kobo[] = {
+    {  0, {0x00, 0x00, 0x00}},
+    { 50, {0x20, 0x20, 0x20}},
+    {100, {0x40, 0x40, 0x40}},
+    {150, {0x60, 0x60, 0x60}},
+    {200, {0x80, 0x80, 0x80}}
```
Comment (MaxKellermann) [inline src/Look/TrailLook.cpp:64]: Why are there 5 items in the array - wouldn't just 2 items result in the same color ramp?

File: src/Renderer/TrailRenderer.cpp [hunk 7]
```diff
@@ -171,15 +171,17 @@ TrailRenderer::Draw(Canvas &canvas, const TraceComputer &trace_computer,
...
           canvas.SelectNullPen();
           canvas.Select(look.trail_brushes[color_index]);
           canvas.DrawCircle({(pt.x + last_point.x) / 2, (pt.y + last_point.y) / 2},
                             look.trail_widths[color_index]);
         } else {
           // positive vario case
 
-          if (settings.type == TrailSettings::Type::VARIO_DOTS_AND_LINES) {
+          if (settings.type == TrailSettings::Type::VARIO_DOTS_AND_LINES ||
+            settings.type == TrailSettings::Type::VARIO_KOBO) {
```
Comment (MaxKellermann) [inline src/Renderer/TrailRenderer.cpp:184]: bad indent

File: NEWS.txt (hunk 2 above)
Comment (ulipo) [reply NEWS.txt]: Yeah, sounds fine. Let me get this PR done and I'll make a new one to clean up trailing spaces :-)

File: src/MapSettings.hpp (hunk 5 above)
Comment (ulipo) [reply src/MapSettings.hpp]: Not sure to understand this (I am more a carpenteer than a coder) but I moved the line to the bottom of the list, I hope it works...

File: src/Look/TrailLook.cpp (hunk 6 above)
Comment (ulipo) [reply src/Look/TrailLook.cpp]: True, I started by tweaking another color ramp and lost contact... Now it's fixed.

File: src/Renderer/TrailRenderer.cpp (hunk 7 above)
Comment (ulipo) [reply src/Renderer/TrailRenderer.cpp]: Fixed

File: src/Dialogs/Settings/Panels/SymbolsConfigPanel.cpp [hunk 8]
```diff
@@ -111,6 +111,7 @@ static constexpr StaticEnumChoice trail_type_list[] = {
     N_("Vario-scaled dots with lines. "
        "Orange to red = climb. Light blue to dark blue = sink. "
        "Zero lift is presented as a yellow line.") },
+  { TrailSettings::Type::VARIO_EINK, N_("Vario E-ink"), N_("E-ink friendly color scheme, lighter and thicker dots means lift while darker and thinner means sink.") },
```
Comment (ulipo) [inline src/Dialogs/Settings/Panels/SymbolsConfigPanel.cpp:114]: I changed the name from VARIO_KOBO to a more generic VARIO_EINK, because I tought there are more eink devices other than Kobo where to run XCSoar (Yotaphone, to name one)

**Comment by lordfolken** [comment]
Please resolve the conversations on github, of the things that you think are fixed.

File: src/MapSettings.hpp (hunk 5 above)
Comment (lordfolken) [reply src/MapSettings.hpp]: Enum is indexed. So actually the index value is saved to the user's profile not the name. If you alter the list instead of appending it, it will change the relationship between index value and trail type.

Comment (ulipo) [reply src/MapSettings.hpp]: Ok, thanks for explaining, now it makes sense to me
